versioned_docs/version-V2/use-cases/private-voting.md

```diff
@@ -0,0 +1,135 @@
+
+### Create a poll
+
+A community coordinator ordApp administrator uses the deployed smart contract to create an on-chain (Ethereum) poll that members can join and vote on.
+
+In the following sample code, the voting contract declares a `createPoll` function that uses the Semaphore base `_createGroup` function:
+
+```ts title="https://github.com/semaphore-protocol/semaphore/contracts/extensions/SemaphoreVoting.sol"
+
+function createPoll(
+    uint256 pollId,
+    address coordinator,
+    uint8 depth
+) public override {
+    require(address(verifiers[depth]) != address(0), "SemaphoreVoting: depth value is not supported");
+
+    // highlight-next-line
+    _createGroup(pollId, depth, 0);
+
+    Poll memory poll;
+
+    poll.coordinator = coordinator;
+
+    polls[pollId] = poll;
+
+    emit PollCreated(pollId, coordinator);
+}
+```
+
+A poll is a Semaphore [group](/docs/guides/groups/) that stores the following:
+
+- A topic to vote on.
+- The public ID of the poll creator.
+- [Semaphore IDs](/docs/guides/identities/) of members who joined the poll.
+
+To create the poll, the administrator calls the smart contract function--for example:
+
+```ts
+SemaphoreVoting.createPoll(pollId, coordinator, depth)
+```
+
+Next, learn how to [register voters](#register-voters) for the poll.
+
+### Register voters
+
+Before a user can register to vote, yourdApp needs to verify membership by checking the user's wallet for the NFT.
+To grant access to the wallet, the user clicks a `Connect wallet` button in thedApp and allows thedApp to check for the NFT.
+Once a member is verified, thedApp provides the following member interactions:
+
+1. [Generate a private identity](#generate-a-private-identity).
+2. [Join a poll](#join-a-poll).
+
+#### Generate a private identity
+
+To generate a private identity, the member completes a form in thedApp UI.
+With the form values and the `@zk-kit/identity` library, thedApp prompts the member to sign a wallet message and then generates the signed private identity.
+The private identity is known only to the member and can be used in future interactions with thedApp.
+
+Next, learn how members [join a poll](#join-a-poll). 
+
+#### Join a poll
+
+Once the member has a private identity for thedApp, the member may select a poll to vote in.
+When the member selects a poll, thedApp does the following:
+
+1. Uses the `zk-kit` library to generate an anonymous Semaphore ID, or _identity commitment_, from the private identity.
+2. Calls a contract function that adds the new Semaphore ID to the on-chain poll.
+
+With a member registered for a poll, learn how thedApp [records votes](#record-votes).
+
+### Record votes
+
+Once members have joined a poll, the coordinator starts the poll to allow voting.
+When a member submits a vote in the UI--for example, selects a radio button--then thedApp does the following:
+
+1. Uses the `semaphore-protocol` library to verify that the voter registered a Semaphore ID to the poll.
```

Comment:
That verification is part of the proof (`@semaphore-protocol/proof`) actually.

Reply:
> That verification is part of the proof (`@semaphore-protocol/proof`) actually.

Ok, I'm looking at semaphore-boilerplate. So that I understand, there is no expectation that the dApp checks group membership before the user signals (and the proof is created), i.e. as long as the user has an identity commitment (even though it could be false), they can signal and generate the proof?

Reply:
I just removed that bullet, but maybe need to provide more explanation for others confused as I was.